### PR TITLE
[MFMA] Support tile size 4x4 version 1

### PR DESCRIPTION
**Pull request by binarman**
This PR enables 4x4 tile size in MFMA based dot operations.

Supported tiled dot is `(4x64) x (64x4) -> (4x4)` in MFMA layout.
However, actual dot operation should have at least 64 output elements, this is a limitation of other layouts appearing during result processing (i.e. blocked layout can not handle tensors smaller than wavesize).

For example, following dots are supported: `(4x64) x (64x16) -> (4x16)`, `(16x64) x (64x4) -> (16x4)` or `(8x64) x (64x8) -> (8x8)`
Following dots are not supporter: `(4x128) x (128x4) -> (4x4)`, `(4x64) x (64x8) -> (4x8)`

This is a first version of dot using mfma 4x4 instructions, with redundancy and reductions.

### CODE REVIEW

**Review by zhanglx13 — APPROVED** [review]

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp [hunk 1]
```diff
@@ -80,115 +80,86 @@ struct DotOpMFMAConversionHelper {
...
-        return rewriter.create<ROCDL::mfma_i32_32x32x8i8>(
-            loc, TypeRange{resType},
-            ValueRange{valA, valB, valC, zeroFlag, zeroFlag, zeroFlag});
-      }
+      return rewriter.create<ROCDL::mfma_i32_32x32x8i8>(
+          loc, TypeRange{resType},
+          ValueRange{valA, valB, valC, zeroFlag, zeroFlag, zeroFlag});
     case MatrixCoreType::INT32_INT8_INT8_INT32_CDNA3:
-      if (mfmaDescr.size == 16) {
-        return rewriter.create<ROCDL::mfma_i32_16x16x32_i8>(
```
Comment (scxiao) [inline lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:182]: @alefimov-amd, it seems like `mfma_i32_16x16x32_i8` is just removed. I also ran it on MI300 and several unit tests failed, for example: `pytest test_core_amd.py::test_dot[64-128-128-2-False-True-none-False-int8-None-160]`

**Review by scxiao — COMMENTED** [review]
@alefimov-amd, it seems like mfma_i32_16x16x32_i8 is just removed. I also ran it on MI300 and several unit tests failed, for example: pytest test_core_amd.py::test_dot[64-128-128-2-False-True-none-False-int8-None-160]

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp (hunk 1 above)
Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:182]: Sorry, you are right

Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:182]: Could you check if #424 fixes the problem?

**Pull request merged**